From 0da4a22022e798fa8624662eefbc97534ad7080f Mon Sep 17 00:00:00 2001
From: techyminati <sinha.aryan03@gmail.com>
Date: Mon, 4 Sep 2023 19:04:18 +0200
Subject: [PATCH] LineageParts: Fix NPE with NFC option being nuked

* Fixes:-
09-04 22:15:47.733  6736  6736 D PartsActivity: Launched with: Intent { cmp=org.lineageos.lineageparts/.PartsActivity (has extras) } action: null component: org.lineageos.lineageparts.PartsActivity part: null fragment: org.lineageos.lineageparts.profiles.SetupActionsFragment
09-04 22:15:47.734  6736  6736 D PartsActivity: Launching fragment: org.lineageos.lineageparts.profiles.SetupActionsFragment
09-04 22:15:47.764  1163  1344 W Choreographer: Frame time is 0.107737 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
09-04 22:15:47.774  6736  6736 E os.lineagepart: Invalid ID 0x00000000.
09-04 22:15:47.774  6736  6736 D AndroidRuntime: Shutting down VM
09-04 22:15:47.775  6736  6736 E AndroidRuntime: FATAL EXCEPTION: main
09-04 22:15:47.775  6736  6736 E AndroidRuntime: Process: org.lineageos.lineageparts, PID: 6736
09-04 22:15:47.775  6736  6736 E AndroidRuntime: android.content.res.Resources$NotFoundException: String resource ID #0x0
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.content.res.Resources.getText(Resources.java:444)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.content.res.Resources.getString(Resources.java:537)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.content.Context.getString(Context.java:705)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at org.lineageos.lineageparts.profiles.actions.item.TriggerItem.getTitle(TriggerItem.java:48)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at org.lineageos.lineageparts.profiles.actions.ItemListAdapter$ItemViewHolder.bind(ItemListAdapter.java:120)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at org.lineageos.lineageparts.profiles.actions.ItemListAdapter$ItemViewHolder.access$300(ItemListAdapter.java:109)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at org.lineageos.lineageparts.profiles.actions.ItemListAdapter.onBindViewHolder(ItemListAdapter.java:82)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7379)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7462)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6319)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6585)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6425)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6421)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4423)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3800)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:795)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at com.google.android.material.appbar.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:99)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:2003)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:866)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1552)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:842)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6980)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onMeasure(DecorView.java:758)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.View.measure(View.java:25833)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3682)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2460)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2739)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2179)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8793)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:845)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:780)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
09-04 22:15:47.775  6736  6736 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
09-04 22:15:47.778  1163  6774 I DropBoxManagerService: add tag=system_app_crash isTagEnabled=true flags=0x2
09-04 22:15:47.778  1163  1855 W ActivityTaskManager:   Force finishing activity org.lineageos.lineageparts/.PartsActivity
09-04 22:15:47.789  1163  1855 W ActivityTaskManager:   Force finishing activity org.lineageos.lineageparts/.profiles.ProfilesSettings
09-04 22:15:47.793  6736  6736 I Process : Sending signal. PID: 6736 SIG: 9
09-04 22:15:47.809  1163  3240 I ActivityManager: Process org.lineageos.lineageparts (pid 6736) has died: fg  TOP
09-04 22:15:47.809   677   677 I Zygote  : Process 6736 exited due to signal 9 (Killed)
09-04 22:15:47.809  1163  1353 I libprocessgroup: Successfully killed process cgroup uid 1000 pid 6736 in 0ms
09-04 22:15:47.810  1163  1720 I WindowManager: WIN DEATH: Window{d7ba756 u0 org.lineageos.lineageparts/org.lineageos.lineageparts.profiles.ProfilesSettings}
09-04 22:15:47.810  1163  1720 W InputManager-JNI: Input channel object 'd7ba756 org.lineageos.lineageparts/org.lineageos.lineageparts.profiles.ProfilesSettings (client)' was disposed without first being removed with the input manager!
09-04 22:15:47.812  1163  1338 I WindowManager: WIN DEATH: Window{117a851 u0 org.lineageos.lineageparts/org.lineageos.lineageparts.PartsActivity}
09-04 22:15:47.813  1163  1338 W InputManager-JNI: Input channel object '117a851 org.lineageos.lineageparts/org.lineageos.lineageparts.PartsActivity (client)' was disposed without first

Change-Id: Iaca0a002ff1e1166e35c09b021c8a6c747475ab6
Signed-off-by: techyminati <sinha.aryan03@gmail.com>
---
 .../lineageparts/profiles/SetupActionsFragment.java         | 3 ---
 .../lineageparts/profiles/actions/item/TriggerItem.java     | 6 +-----
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/org/lineageos/lineageparts/profiles/SetupActionsFragment.java b/src/org/lineageos/lineageparts/profiles/SetupActionsFragment.java
index cc45ad3..4628d93 100644
--- a/src/org/lineageos/lineageparts/profiles/SetupActionsFragment.java
+++ b/src/org/lineageos/lineageparts/profiles/SetupActionsFragment.java
@@ -197,9 +197,6 @@ public class SetupActionsFragment extends SettingsPreferenceFragment
             if (DeviceUtils.deviceSupportsBluetooth()) {
                 mItems.add(generateTriggerItem(TriggerItem.BLUETOOTH));
             }
-            if (DeviceUtils.deviceSupportsNfc(context)) {
-                mItems.add(generateTriggerItem(TriggerItem.NFC));
-            }
         }
 
         // connection overrides
diff --git a/src/org/lineageos/lineageparts/profiles/actions/item/TriggerItem.java b/src/org/lineageos/lineageparts/profiles/actions/item/TriggerItem.java
index ae911a6..616ea73 100644
--- a/src/org/lineageos/lineageparts/profiles/actions/item/TriggerItem.java
+++ b/src/org/lineageos/lineageparts/profiles/actions/item/TriggerItem.java
@@ -29,7 +29,7 @@ public class TriggerItem extends Item {
     public static final int WIFI = 0;
     public static final int BLUETOOTH = 1;
     // not in Profile.TriggerType, but we need it.
-    public static final int NFC = 2;
+  //  public static final int NFC = 2;
 
     private final Profile mProfile;
     private final int mTriggerType;
@@ -61,11 +61,7 @@ public class TriggerItem extends Item {
         }
 
         if (sb.length() == 0) {
-            if (mTriggerType == NFC) {
-                return context.getString(R.string.no_triggers_configured_nfc);
-            } else {
                 return context.getString(R.string.no_triggers_configured);
-            }
         }
 
         return sb.toString();
-- 
2.34.1
